scripts/coccicheck

```diff
@@ -201,6 +212,7 @@ coccinelle () {
 	run_cmd_parmap $SPATCH -D $MODE   $FLAGS --cocci-file $COCCI $OPT $OPTIONS || exit 1
     fi
 
+    MODE=report
```

Comment:
what does it doing here ?

Reply:
The covers the case when suppose a script has no report mode and instead patch mode(available)
is run. This is fine behavior.
But the problem is that next upcoming script to be run also uses the previous propagated
used mode which was patch mode and not report mode.
So, the rule is to keep report mode as the top priority and be less verbose in warnings/reports. Therefore to fulfil such a condition I overwrite MODE variable with report, so that upcoming
script to be used runs in report mode(if available).

Now, as @JuliaLawall suggested it would be better to support at the very least report mode in
all coccinelle scripts in the current and future scripts. That would be really helpful in case we
add coccicheck in the automated CI.